src/browserbehavior/ExtendedBrowserBehavior.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { BrowserBehavior } from '..';
+
+/**
+ * This interface contains methods on {@link DefaultBrowserBehavior} that were
+ * incorrectly omitted from {@link BrowserBehavior}, plus new methods that
+ * were added since the last major version.
+ *
+ * Members of this interface can change without a major version bump to accommodate new browser
+ * bugs and capabilities. If you extend this type, you might need to rework your code for new minor
+ * versions of this library.
+ */
+export default interface ExtendedBrowserBehavior extends BrowserBehavior {
+  requiresResolutionAlignment(width: number, height: number): [number, number];
```

Comment:
Can these methods added as optional methods in the `BrowserBehavior` itself? I think these methods are now in both the interfaces right?

Reply:
`DDC` is written to only work with a `DefaultBrowserBehavior`. It requires these methods, and generalizing the `DDC` constructor to allow the behavior object to be specified requires either making the argument a `DBB` or introducing an interface. 

Making them optional in the existing public interface doesn't actually help much, because not implementing the methods means that DDC doesn't know what to do.

We _could_ make an additional interface called `MoreBrowserBehavior`, and make the argument to `DDC` be of type `BrowserBehavior & MoreBrowserBehavior`, but I chose to just extend the type instead.

You could view `ExtendedBrowserBehavior` as a shorter name for `TheKindOfBrowserBehaviorThatOurDeviceControllerActuallyRequiresToFunction`.

Reply:
Ok got it! Thanks for the explanation.